File: julia/rna-transcription/rna-transcription.jl
const RNA_DATA = Dict('C' => 'G', 'G' => 'C', 'T' => 'A', 'A' => 'U')

function to_rna(dna)
   
    if map(x -> x ∉ keys(RNA_DATA), collect(dna)) |> any
        throw(ErrorException("Invalid nucleotide"))
    end
    map(x -> RNA_DATA[x], dna)

end

File: elixir/rna-transcription/lib/rna_transcription.ex
defmodule RnaTranscription do
  @doc """
  Transcribes a character list representing DNA nucleotides to RNA

  ## Examples

  iex> RnaTranscription.to_rna('ACTG')
  'UGAC'
  """
  @spec to_rna([char]) :: [char]
  def to_rna(dna) do
    Enum.map(dna,fn x ->RnaTranscription.process(x) end)
  end


  @spec process(any) :: 32 | 65 | 67 | 71 | 85
  def process(?G) do ?C end
  def process(?C) do ?G end
  def process(?T) do ?A end
  def process(?A) do ?U end
  def process(_) do 0 end


end
